src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -82,9 +82,15 @@
     abstract public SamReader open(final File file);
 
     public SamReader open(final Path path) {
-        final SamInputResource r = SamInputResource.of(path, getPathWrapper());
+        return open(path, getPathWrapper(), Function.identity());
```

Comment:
Actually since we've now moved to passing the wrappers explicitly to open, I suggest we remove setPathWrapper. The open call without wrappers will not apply any wrapper, behaving like the other methods that take optional wrapper arguments.